Cython/Compiler/ExprNodes.py

```diff
@@ -7047,6 +7029,32 @@ def analyse_as_type_attribute(self, env):
                     error(self.pos, "%s not a known value of %s" % (self.attribute, type))
         return None
 
+    def _create_ubcm_entry(self, type, entry, env):
+        # Create a temporary entry describing the unbound C method in `entry`
+        # as an ordinary function.
+        if entry.func_cname and entry.type.op_arg_struct is None:
+            cname = entry.func_cname
+            if entry.type.is_static_method or (
+                    env.parent_scope and env.parent_scope.is_cpp_class_scope):
+                ctype = entry.type
+            elif type.is_cpp_class:
+                error(self.pos, "%s not a static member of %s" % (entry.name, type))
+                ctype = PyrexTypes.error_type
```

Comment:
This seems worth an additional error test, given that it now extends to more cases. See the examples in `tests/errors/`.

Reply:
I'm actually not sure when exactly this code will be reached. For example, I can trigger it with:

```cython
cdef extern from *:
    """
    struct Foo
    {
        struct Bar {
        };
    };
    """
    cdef cppclass Foo:
        Foo()
        cppclass Bar:
            Bar()
        Bar()

Foo.Bar()  # no error without this
```

```
> Bar not a static member of Foo
```

But it feels like this should either have been caught somewhere else; or raise a different kind of error. Any suggestions here on what precisely we should be testing?